test/health_check_test_case.py

```diff
@@ -84,6 +89,35 @@ def test_health_endpoint_keys(self):
                         self.assertEqual(200, response.status_code)
                         self.assertEqual(expected_response, response.json())
 
+    @mock_s3
+    @mock_sts
+    @mock_sqs
+    def test_cached_health(self):
+        storage_service = StorageService()
+        storage_service.create_bucket()
+        with ResponsesHelper() as helper:
+            helper.add_passthru(self.base_url)
+            response = requests.get(self.base_url + '/health/cached')
+            self.assertEqual(500, response.status_code)
+
+        self._create_mock_queues()
+        endpoint_states = self._make_endpoint_states(self.endpoints)
+        app = load_app_module(self.lambda_name())
+        with ResponsesHelper() as helper:
+            helper.add_passthru(self.base_url)
+            self._mock_service_endpoints(helper, endpoint_states)
+            with mock.patch.dict(os.environ, AWS_DEFAULT_REGION='us-east-1'):
+                app.generate_health_object(MagicMock(), MagicMock())
+                response = requests.get(self.base_url + '/health/cached')
+                self.assertEqual(200, response.status_code)
+
+        future_time = time.time() + 3 * 60
+        with ResponsesHelper() as helper:
+            helper.add_passthru(self.base_url)
+            with mock.patch('time.time', new=lambda: future_time):
+                response = requests.get(self.base_url + '/health/cached')
+                self.assertEqual(500, response.status_code)
```

Comment:
Can you assert the body here? It should contain "Cached health object is stale". I don't remember if raising an exception produces a Chalice response whose body is the message from the exception. But if that doesn't work, you can always `return Response(body="Cached health object is stale", status_code=500)` instead of raising.